core/src/main/java/org/apache/brooklyn/core/entity/EntityRelations.java

```diff
@@ -0,0 +1,160 @@
+        "entity", "entities", Entity.class, "has_policy", 
+        "policy", "policies", Policy.class, "policy_for");
+    /** Inverse of {@link #HAS_POLICY}. */
+    public static final Relationship<Policy,Entity> POLICY_FOR = HAS_POLICY.getInverseRelationship();
+
+    // ----
+    
+    // TODO replace by relations stored in catalog when catalog supports arbitrary types
+    private static Map<String,Relationship<? extends BrooklynObject, ? extends BrooklynObject>> KNOWN_RELATIONSHIPS = MutableMap.of();
+    private static void addRelationship(Relationship<? extends BrooklynObject, ? extends BrooklynObject> r) {
+        KNOWN_RELATIONSHIPS.put(r.getRelationshipTypeName(), r);
+        if (r.getInverseRelationship()!=null) {
+            KNOWN_RELATIONSHIPS.put(r.getInverseRelationship().getRelationshipTypeName(), r.getInverseRelationship());
+        }
+    }
+    static {
+        addRelationship(MANAGER_OF);
+        addRelationship(GROUP_CONTAINS);
+        addRelationship(HAS_TARGET);
+        addRelationship(HAS_POLICY);
+    }
+    
+    /** Find the typed Relationship instance for the given relationship name, if known;
+     * behaviour is not guaranteed by the API if not known (hence the Beta marker),
+     * it may fail fast or return null or create a poor-man's relationship instance. 
+     */
+    @Beta
+    public static Relationship<? extends BrooklynObject, ? extends BrooklynObject> lookup(ManagementContext mgmt, String relationshipTypeName) {
+        if (relationshipTypeName==null) return null;
+        Relationship<? extends BrooklynObject, ? extends BrooklynObject> result = KNOWN_RELATIONSHIPS.get(relationshipTypeName);
+        if (result!=null) return result;
+        
+        /* TODO ultimately we'd like to support arbitrary relationships via persistence and lookup against the catalog;
+         * however for now, so that we can persist nicely (without catalog items for relationships) 
+         * we are smart about the relationships defined here, and we return a poor-man's version for items elsewhere.
+         * 
+         * for now, a poor-man's relationship; if not in catalog ultimately we should fail. */
+        return Relationships.newRelationshipOneway("source", "sources", BrooklynObject.class, relationshipTypeName, "target", "targets", BrooklynObject.class);
+    }
+    
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    public static <T extends BrooklynObject> Set<Relationship<? super T,? extends BrooklynObject>> getRelations(T source) {
+        return ((BrooklynObjectInternal)source).relations().getLocalBackingStore().getRelationships();
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    public static <T extends BrooklynObject,U extends BrooklynObject> Set<U> getRelationships(Relationship<? super T,U> relationship, T source) {
+        return ((BrooklynObjectInternal)source).relations().getLocalBackingStore().getRelations(relationship);
```

Comment:
Seems you've got the backing store method names swapped. `getRelations` from above calls `getRelationships` and `getRelationships` here delegates to `getRelations`